File: android/src/main/java/com/jimmydaddy/imagemarker/base/SaveFormat.java
package com.jimmydaddy.imagemarker.base;

public enum SaveFormat {
  PNG("png"),
  JPG("jpg"),
  BASE64("base64");

  private final String value;

  private SaveFormat(String value) {
    this.value = value;
  }

  public String getValue() {
    return value;
  }

  static public SaveFormat getFormat(String format) {
    switch (format) {
      case "jpg":
      case "JPG":
      case "JPEG":
      case "jpeg":
        return JPG;
      case "base64":
      case "BASE64":
        return BASE64;
      default:
        return PNG;
    }
  }
}

File: android/src/main/java/com/jimmydaddy/imagemarker/base/SaveFormat.kt
package com.jimmydaddy.imagemarker.base

enum class SaveFormat(val value: String) {
  PNG("png"), JPG("jpg"), WEBP("webp"), BASE64("base64");

  companion object {
    fun getFormat(format: String?): SaveFormat {
      return when (format) {
        "jpg", "JPG", "JPEG", "jpeg" -> JPG
        "base64", "BASE64" -> BASE64
        "png", "PNG" -> PNG
        "webp", "WEBP" -> WEBP
        null -> JPG
        else -> throw MarkerError(ErrorCode.INVALID_PARAMS, "Unsupported saveFormat: $format")
      }
    }
  }
}
